backend/src/controllers/dish.ts

```diff
@@ -100,29 +74,70 @@ export const getDishes = async (req: Request, res: Response) => {
             message: 'Error when mapping dishes to transactions',
             statusCode: 500,
         })
-        res.status(500).json({ error: 'internal_server_error' })
-        return
+        return res.status(500).json({ error: 'internal_server_error' })
+    }
+
+    let userDishes
+    if (withTransactions !== 'true') {
+        try {
+            if (borrowed == 'true') {
+                userDishes = getAllUserDishesInUse(userClaims, allDishes, dishTransMap)
+            } else {
+                userDishes = getAllUserDishes(userClaims, allDishes, dishTransMap)
+            }
+        } catch (e) {
+            Logger.error({
+                error: e,
+                message: 'error when getting user dishes',
+                statusCode: 500
+            })
+            return res.status(500).json({ error: 'internal_server_error' })
+        }
+        return res.status(200).json({ userDishes })
     }
 
+    let allDishesVM = <Array<any>>[]
     try {
-        allDishesVM = mapToDishVM(rawDishData, dishTransMap)
-        Logger.info({
-            message: 'Mapped transactions to view model',
-        })
+        allDishesVM = mapToDishVM(allDishes, dishTransMap)
+        Logger.info({ message: 'Mapped transactions to view model' })
     } catch (e) {
         Logger.error({
             error: e,
             message: 'Error when mapping transactions to view model',
             statusCode: 500,
         })
-        res.status(500).json({ error: 'internal_server_error' })
-        return
+        return res.status(500).json({ error: 'internal_server_error' })
+    }
+    Logger.info({ message: 'Sending dish data with transactions' })
+
+    if (all == 'true') {
+        if (!verifyIfUserAdmin(userClaims)) {
+            Logger.error({
+                message: 'User is not admin',
+                statusCode: 403
+            })
+            return res.status(403).json({ error: 'forbidden' })
+        }
+
+        req.log.info('sending all dishes to admin')
+        return res.status(200).json({ allDishesVM })
+    }
+
+    let userDishesVM
+    try {
+        if (borrowed !== 'true') {
+            userDishesVM = getAllUserDishesVM(userClaims, allDishesVM, dishTransMap)
+        } else {
+            userDishesVM = getAllUserDishesInUse(userClaims, allDishesVM, dishTransMap)
+        }
+    } catch (e) {
+        Logger.error({
+            error: e,
+            message: 'error when getting user dishes view model',
+            statusCode: 500
+        })
+        return res.status(500).json({ error: 'internal_server_error' })
     }
 
-    Logger.info({
-        message: 'Sending dish data with transactions',
-    })
-    // send response
-    res.status(200).json({ dishes: allDishesVM })
-    return
+    return res.status(200).json({ userDishesVM })
```

Comment:
Can we keep consistent naming?
```suggestion
    return res.status(200).json({ dishes: userDishesVM })
```